lib/Models/CkanCatalogGroup.ts

```diff
@@ -84,9 +84,13 @@ export class CkanServerStratum extends LoadableStratum(CkanCatalogGroupTraits) {
         .segment("api/3/action/package_search")
         .addQuery({ start: 0, rows: 1000, sort: "metadata_created asc" });
 
-      Object.keys(filterQuery).forEach((key: string) =>
-        uri.addQuery(key, filterQuery[key])
-      );
+      if (filterQuery.toString().includes("fq=")) {
+        uri.query(uri.query() + "&" + filterQuery.toString());
+      } else {
```

Comment:
If the `filterQuery` trait can have string elements, it needs to be reflected in the `Trait`

https://github.com/TerriaJS/terriajs/blob/6e4d4df9dc368b155d283abd48bbe0c788960616/lib/Traits/CkanCatalogGroupTraits.ts#L25-L43

Reply:
At the moment it only allows `JsonObject` elements

Reply:
A possible change could be:
```ts
  filterQuery?: {
    [key: string]: JsonValue | string;
  }[]
```

Reply:
If `filterQuery` can only have `JsonObject` elements. An element like "fq=+blahblah" is considered valid JsonObject.

Reply:
E.g.  filterQuery?: Object[] = ["fq=+blahblah"];

Reply:
String isn't a valid JsonObject;

```
export interface JsonObject {
  [key: string]: JsonValue;
}
```

But I just realised that my suggestion is totally wrong! I meant something like this:

```filterQuery?: (JsonObject | string)[] = [
    "Some string",
    {
      fq:
        '(res_format:(czml OR CZML OR geojson OR GeoJSON OR WMS OR wms OR kml OR KML OR kmz OR KMZ OR WFS OR wfs OR CSV-GEO-AU OR csv-geo-au OR "Esri REST"))'
    }
  ];
```

Reply:
But also, why do we need a `string`?
 
Isn't `["fq=+blahblah"]` equivalent to `{fq: "blahblah"}`

Reply:
@nf-s You are right. I made a mistake by using Object instead of JsonObject in the example
 ```filterQuery?: Object[] = ["fq=+blahblah"];```
which is valid but useless. Has adopted your recommendation, which will cause other improvement. See the new commit.